cli/src/main/java/com/devonfw/tools/ide/io/IniParserImpl.java

```diff
@@ -0,0 +1,127 @@
+package com.devonfw.tools.ide.io;
+
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import com.devonfw.tools.ide.context.IdeContext;
+
+/**
+ * Implementation of {@link IniParser} preserves order of sections and properties between reading and writing
+ */
+public class IniParserImpl implements IniParser {
+
+  LinkedHashMap<String, LinkedHashMap<String, String>> iniMap;
+  IdeContext context;
```

Comment:
Avoid keeping such state inside the data-structure for an `.ini` file and rather make it stateless.